inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/ContextManager.java

```diff
@@ -59,4 +63,10 @@ public void attachContextToThread(Thread thread) {
             context.attach();
         }
     }
+
+    @Override
+    public IInspectitContext enterNewContext() {
+        return InspectitContext.createFromCurrent(commonTagsManager, configProvider.getCurrentConfig().getDataProperties(), IS_OPEN_CENSUS_ON_BOOTSTRAP);
```

Comment:
update configuration will update the `currentConfig` instance in the `InstrumentationConfigurationResolver`, which means that this getter here is not thread-safe.. However, maybe this is a fine trade-off, to see configuration updates later but have no concurrent data structure to protect it.. I am just wondering if we can get into some strange situations if this is the case?

Reply:
The currentConfiguration is never changed but only replaced. This means that we don't actually ahve a race condition here. The semantic is that we simply take the most recent configuration for data propagation when a context is created. We never change this configuration for an already opened context.